### ISSUE TITLE
How do you deal with the problem of overfitting?

### ISSUE THREAD

**Issue by renyuzh** [halwai/bird_classification#9]
I train the model at the CUB dataset serverl times, but the result is always overfitting even if some data augmentation, do you  met the overfitting problem and how do you deal with it?

**Comment by halwai** [comment]
All neural network models face the problem of overfitting on the training data. Some techniques are quite popular to avoid these:
1) Dropout 
2) Batch Normalization
3) Adjusting Learning Rates
4) Regularization terms (regularization of weights)
5) Keeping an eye out on learning by using validation set accuracy

All of these methods are quite well know and are used in most current-publications as a direct result. Please google for more theoretical or practical aspects. I guess I tried combining (3),(5). Though the code can be easily extended to suit other points as well.
Not all of these methods are exclusive and a combination of them may help better.
